In [25]:
import requests
from bs4 import BeautifulSoup

Let's crawl chi 2015 proceeding. Use 'single page view' layout instead of 'tab view'

In [9]:
url = "https://dl.acm.org/citation.cfm?id=2702123&preflayout=flat"

In [35]:
headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'}
s = requests.Session()
resp = s.get(url, headers=headers)

In [36]:
soup = BeautifulSoup(resp.text, 'html.parser')

In [37]:
links = soup.find_all("a", {"name": "FullTextPDF"})

In [38]:
len(links)

484

There are 484 pdf links. To download each link

In [51]:
pdf_loc = links[0].get('href')

In [52]:
pdf_url = "https://dl.acm.org/" + pdf_loc

In [43]:
resp_pdf = s.get(pdf_url, headers=headers)

In [44]:
print(resp_pdf)

<Response [200]>


In [53]:
with open('./{}.pdf'.format(get_id(pdf_loc)), 'wb') as f:
    f.write(resp_pdf.content)

In [49]:
def get_id(s):
    params = s.split("?")[1]
    params = params.split("&")
    return params[0].split("=")[1]

TODO: 
What if ACM library block our lab's IP?